src/navigation/navigation.tsx

```diff
@@ -1,21 +1,58 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
+import { useDispatch } from 'react-redux'
+import { useAppSelector } from 'src/hooks'
 import { createStackNavigator } from '@react-navigation/stack'
 import { SignIn, Home } from 'src/screens'
-import { NavigationScreen } from 'src/common/enums'
+import { NavigationScreen, StorageKey } from 'src/common/enums'
+import { storage } from 'src/services'
+import { getCurrentUser } from 'src/store/actions'
 
 const Stack = createStackNavigator()
 
 const Navigation: React.FC = () => {
+  const { user } = useAppSelector(({ auth }) => ({
+    user: auth.user
+  }))
+
+  const dispatch = useDispatch()
+
+  const [isFirstLoading, setIsFirstLoading] = useState(true)
+  const [token, setToken] = useState<string | null>(null)
+
+  const hasToken = Boolean(token)
+  const hasUser = Boolean(user)
+
+  const isUserExistsAndNotLoaded = hasToken && !hasUser
+
+  useEffect(() => {
+    storage
+      .getItem(StorageKey.TOKEN)
+      .then(setToken)
+      .then(() => setIsFirstLoading(false))
+  }, [])
+
+  useEffect(() => {
+    if (token) {
+      dispatch(getCurrentUser())
+    }
+  }, [token])
+
+  if (isFirstLoading || isUserExistsAndNotLoaded) {
+    return null
```

Comment:
I'm not sure, maybe we need to replace user to other screen in this case or displaying preloader?

Reply:
Yes, we can. But for now `null` is just a replacement for preloader.